vmc/resource_vmc_srm_nodes.go

```diff
@@ -0,0 +1,159 @@
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 13),
+				Description:  "The custom extension suffix for SRM must contain 13 characters or less, be composed of letters, numbers, ., - characters only. The suffix is appended to com.vmware.vcDr- to form the full extension key. ",
+			},
+			"srm_instance": {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceSRMNodesCreate(d *schema.ResourceData, m interface{}) error {
+
+	connector := (m.(*ConnectorWrapper)).Connector
+
+	siteRecoverySrmNodesClient := draas.NewDefaultSiteRecoverySrmNodesClient(connector)
+
+	srmExtensionKeySuffix := d.Get("srm_node_extension_key_suffix").(string)
+	orgID := (m.(*ConnectorWrapper)).OrgID
+	sddcID := d.Get("sddc_id").(string)
+
+	provisionSrmConfigParam := &model.ProvisionSrmConfig{
+		SrmExtensionKeySuffix: &srmExtensionKeySuffix,
+	}
+
+	task, err := siteRecoverySrmNodesClient.Post(orgID, sddcID, provisionSrmConfigParam)
+
+	if err != nil {
+		return fmt.Errorf("Error while activating site recovery instance for sddc %s: %v", sddcID, err)
+	}
+
+	taskID := task.ResourceId
+	d.SetId(*taskID)
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		tasksClient := draas.NewDefaultTaskClient(connector)
+		task, err := tasksClient.Get(orgID, task.Id)
+		if err != nil {
+			if err.Error() == (errors.Unauthenticated{}.Error()) {
+				log.Print("Auth error", err.Error(), errors.Unauthenticated{}.Error())
+				err = (m.(*ConnectorWrapper)).authenticate()
+				if err != nil {
+					return resource.NonRetryableError(fmt.Errorf("Error authenticating in CSP: %s", err))
+				}
+				return resource.RetryableError(fmt.Errorf("Instance creation still in progress"))
+			}
+			return resource.NonRetryableError(fmt.Errorf("Error describing instance: %s", err))
+
+		}
+		if *task.Status != "FINISHED" {
+			return resource.RetryableError(fmt.Errorf("Expected instance to be created but was in state %s", *task.Status))
+		}
+		return resource.NonRetryableError(resourceSRMNodesRead(d, m))
+	})
+	return nil
+
+}
+
+func resourceSRMNodesRead(d *schema.ResourceData, m interface{}) error {
+
+	connector := (m.(*ConnectorWrapper)).Connector
+	orgID := (m.(*ConnectorWrapper)).OrgID
+	sddcID := d.Get("sddc_id").(string)
+
+	siteRecoveryClient := draas.NewDefaultSiteRecoveryClient(connector)
+	siteRecovery, err := siteRecoveryClient.Get(orgID, sddcID)
+	if err != nil {
+		if err.Error() == errors.NewNotFound().Error() {
+			log.Printf("SRM information for SDDC with ID %s not found", sddcID)
+			d.SetId("")
+			return fmt.Errorf("SRM information for SDDC with ID %s not found", sddcID)
+		}
+		return fmt.Errorf("Error while getting SRM instance information for SDDC with ID %s : %v", sddcID, err)
+	}
+	srmExtensionKey := d.Get("srm_node_extension_key_suffix").(string)
+	srm_node := map[string]string{}
+	var i int
+	for i = 0; i < len(siteRecovery.SrmNodes); i++ {
```

Comment:
int declaration on line 119 can be removed by changing for to "for i := 0.."

Reply:
Done